File: OJ/usaco/s1/s11/gift1.pas
{
ID:lijianx1
PROG: gift1
LANG: PASCAL
}
program gift1;
type node=record
	 name:string;
	 money,last:longint;
     end;
var a:array[1..10] of node;
    i,j,k,m,sum,n:longint;
    s:string;

begin
    assign(input,'gift1.in'); reset(input);
    readln(n);
    for i:=1 to n do begin
	readln(a[i].name);
	a[i].money:=0;
    end;
    for i:=1 to n do begin
	readln(s);
	for j:=1 to n do if s=a[j].name then break;
	readln(sum,m); a[j].last:=sum;
	a[j].money:=a[j].money+sum;
	if m<>0 then a[j].money:=a[j].money-(sum div m)*m;
	for j:=1 to m do begin
	    readln(s);
	    for k:=1 to n do if s=a[k].name then break;
	    a[k].money:=a[k].money+sum div m;
	end;
    end;
    close(input);
    assign(output,'gift1.out'); rewrite(output);
    for i:=1 to n do writeln(a[i].name,' ',a[i].money-a[i].last);
    close(output);
end.

File: OJ/usaco/cpp/gift1.cpp
/*
ID:oifox20071
LANG:C++
PROG:gift1
*/
#include <iostream>
#include <string>
using namespace std;
const
	int maxn = 10;
int money[maxn], last[maxn], n;
string name[maxn];

int find(string a)
{
	for (int i = 0; i < n; ++ i)
		if (a == name[i]) return i;
}
int main()
{
	freopen("gift1.in", "r", stdin); freopen("gift1.out", "w", stdout);
	cin >> n;
	for (int i = 0; i < n; ++ i)
	{
		money[i] = 0;
		cin >> name[i];
	}
	string cur;
	int sum, m;
	for (int i = 0; i < n; ++ i)
	{
		cin >> cur >> sum >> m;
		int k = find(cur);
		last[k] = sum;
		money[k] += sum;
		if (m != 0) money[k] -= (sum / m) * m;
		for (int j = 0; j < m; ++ j)
		{
			cin >> cur;
			int p = find(cur);
			money[p] += sum / m;
		}
	}
	for (int i = 0; i < n; ++ i)
		cout << name[i] << " " << money[i] - last[i] << endl;
	return 0;
}
